pkg/api/admin/api/config.go

```diff
@@ -0,0 +1,142 @@
+package api
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+
+	"github.com/satori/go.uuid"
+	"k8s.io/client-go/tools/clientcmd/api/v1"
+)
+
+type Config struct {
+	// configuration of VMs in ARM template
+	ImageOffer     *string `json:"imageOffer,omitempty"`
+	ImagePublisher *string `json:"imagePublisher,omitempty"`
+	ImageSKU       *string `json:"imageSku,omitempty"`
+	ImageVersion   *string `json:"imageVersion,omitempty"`
+
+	SSHKey *rsa.PrivateKey `json:"sshKey,omitempty"`
```

Comment:
There should be no keys or secrets in the admin API.  We neither want to risk accidentally returning these to an SRE, nor make it theoretically expressible to set these to a value known by an SRE.  Please remove all keys, secrets, passwords, kubeconfigs across the board.  In the future there may be a separate action which returns a kubeconfig to an SRE, under separate audit rules.

Reply:
How will we be able to rotate them if we don't expose them in this API? How will Charle's key rotation test work with this API then?

Reply:
(brain dump) I think we need something different for this. Like "Rotate keys" and we could have something like validationAPI which would return if backend objects are valid. If we rotated sshKey, it would return ssh key type and validity. overall as SRE I would like to check integrity of my config after action was done. So some callback is needed without sensitive data.

Reply:
Yeah, makes sense. We discussed this already with @jim-minter but I hadn't fully grasped how it was going to look like.

Reply:
@jim-minter should I also remove ServicePrincipalProfile and AzProfile?